erpnext/patches/v14_0/delete_agriculture_doctypes.py

```diff
@@ -0,0 +1,19 @@
+import frappe
+
+
+def execute():
+	frappe.delete_doc("Module Def", "Agriculture", ignore_missing=True, force=True)
+
+	frappe.delete_doc("Workspace", "Agriculture", ignore_missing=True, force=True)
+
+	reports = frappe.get_all("Report", {"module": "agriculture", "is_standard": "Yes"}, pluck='name')
+	for report in reports:
+		frappe.delete_doc("Report", report, ignore_missing=True, force=True)
+
+	dashboards = frappe.get_all("Dashboard", {"module": "agriculture", "is_standard": 1}, pluck='name')
+	for dashboard in dashboards:
+		frappe.delete_doc("Dashboard", dashboard, ignore_missing=True, force=True)
+
+	doctypes = frappe.get_all("DocType", {"module": "agriculture", "custom": 0}, pluck='name')
+	for doctype in doctypes:
+		frappe.delete_doc("DocType", doctype, ignore_missing=True)
```

Comment:
Tested locally.

Few changes:
1. Domain Agriculture remains in DB, it should be deleted IMO. 
2. This patch shouldn't run if the agriculture **app** is already installed?

Reply:
everything else mostly looks good 👍

Reply:
> Domain Agriculture remains in DB, it should be deleted IMO.

Yes. We should remove it.

> This patch shouldn't run if the agriculture app is already installed?

When a doctype is deleted, the table associated with the doctype remains as is. When an app is installed, it will try to recreate modules, doctypes etc. In some cases presence of existing records will create a problem. It is better to delete these records even if the app is installed.

Reply:
If this patch runs after someone installed the app then won't it cause deletion of doctypes?

Or we are relying on them coming back because doctype migration happens after patches?

Reply:
This patch will delete all the doctypes for all the users.

Users who wants to use agriculture will install the app separately after upgrading ERPNext. When agriculture app is installed, it will re-create all the doctypes, module defs etc.